analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs

```diff
@@ -63,48 +65,54 @@ public TestSetup(string testCase, SonarDiagnosticAnalyzer analyzer, IEnumerable<
-
-            // CodeBlockAnalysisContext
-            // S5693 - MAIN and TEST
-            new TestSetup("GetHashCodeEqualsOverride.cs", new GetHashCodeEqualsOverride()),
-
-            // SymbolAnalysisContext
-            // S2953 - MAIN only
-            new TestSetup("DisposeNotImplementingDispose.cs", new DisposeNotImplementingDispose()),
-            // S1694 - MAIN only
-            new TestSetup("ClassShouldNotBeAbstract.cs", new ClassShouldNotBeAbstract()),
-        });
+        private readonly List<TestSetup> testCases = new(
+            new[]
+            {
+                // SyntaxNodeAnalysisContext
+                // S3244 - MAIN and TEST
+                new TestSetup("AnonymousDelegateEventUnsubscribe.cs", new AnonymousDelegateEventUnsubscribe()),
+                // S2699 - TEST only
+                new TestSetup(
+                    "TestMethodShouldContainAssertion.NUnit.cs",
+                    new TestMethodShouldContainAssertion(),
+                    TestMethodShouldContainAssertionTest.AdditionalTestReferences(NuGetMetadataReference.NUnit(Constants.NuGetLatestVersion))),
+
+                // SyntaxTreeAnalysisContext
+                // S3244 - MAIN and TEST
+                new TestSetup("AsyncAwaitIdentifier.cs", new AsyncAwaitIdentifier()),
+
+                // CompilationAnalysisContext
+                // S3244 - MAIN and TEST
+                new TestSetup(
+                    @"Hotspots\RequestsWithExcessiveLength.cs",
+                    new RequestsWithExcessiveLength(AnalyzerConfiguration.AlwaysEnabled),
+                    RequestsWithExcessiveLengthTest.GetAdditionalReferences()),
+
+                // CodeBlockAnalysisContext
+                // S5693 - MAIN and TEST
+                new TestSetup("GetHashCodeEqualsOverride.cs", new GetHashCodeEqualsOverride()),
+
+                // SymbolAnalysisContext
+                // S2953 - MAIN only
+                new TestSetup("DisposeNotImplementingDispose.cs", new DisposeNotImplementingDispose()),
+                // S1694 - MAIN only
+                new TestSetup("ClassShouldNotBeAbstract.cs", new ClassShouldNotBeAbstract()),
+            });
 
         [TestMethod]
         public void WhenShouldAnalysisBeDisabledReturnsTrue_NoIssueReported()
         {
             SonarAnalysisContext.ShouldExecuteRegisteredAction = (diags, tree) => false;
-
             try
             {
                 foreach (var testCase in testCases)
                 {
                     // ToDo: We should find a way to ack the fact the action was not run
-                    OldVerifier.VerifyNoIssueReported(testCase.Path,
-                                                   testCase.Analyzer,
-                                                   ParseOptionsHelper.FromCSharp8,
-                                                   testCase.AdditionalReferences);
+                    testCase.Verifier
+                        .AddPaths(testCase.Path)
```

Comment:
I think that we can simplify this further by adding the `Path` and the references in the `TestSetup` constructor. We already have these when the builder is created.

Then we will be able to remove the `AdditionalReferences` since it will no longer be needed.